src/util/draftmancerUtil.ts

```diff
@@ -0,0 +1,143 @@
+import { detailsToCard } from '../client/utils/cardutil';
+import { CardDetails } from '../datatypes/Card';
+import { DraftmancerPick } from '../datatypes/Draft';
+import { Decklist, Pick } from '../datatypes/Draftmancer';
+import { getReasonableCardByOracle } from './carddb';
+import { deckbuild } from './draftbots';
+import { getCardDefaultRowColumn, setupPicks } from './draftutil';
+
+export const upsertCardAndGetIndex = (cards: CardDetails[], oracleId: string): number => {
+  const card = getReasonableCardByOracle(oracleId);
+  const index = cards.findIndex((c) => c.oracle_id === oracleId);
+
+  if (index === -1) {
+    cards.push(card);
+    return cards.length - 1;
+  }
+
+  return index;
+};
+
+export const getPicksFromPlayer = (
```

Comment:
Functions all extracted as is, apart from a couple renames for clarity which I think was mainly cards -> cardDetails